mailbox/noise.go

```diff
@@ -449,6 +453,44 @@ func (h *handshakeState) writeMsgPattern(w io.Writer, mp MessagePattern) error {
 			return err
 		}
 
+	case HandshakeVersion1:
+		var payload []byte
+		switch mp.ActNum {
+		case act1, act3:
+		case act2:
+			payload = h.payloadToSend
+
+			// The total length of each message payload including
+			// the MAC size payload exceed the largest number
+			// encodable within a 32-bit unsigned integer.
+			if len(payload) > math.MaxUint32 {
+				return ErrMaxMessageLengthExceeded
+			}
+
+			// The full length of the packet is only the packet
+			// length, and does NOT include the MAC.
+			fullLength := uint32(len(payload))
+
+			var pktLen [4]byte
+			binary.BigEndian.PutUint32(pktLen[:], fullLength)
+
+			// First, generate the encrypted+MAC'd length prefix for
+			// the packet.
+			_, err := buff.Write(h.EncryptAndHash(pktLen[:]))
+			if err != nil {
+				return err
+			}
+
+		default:
+			return fmt.Errorf("unknown act number: %d", mp.ActNum)
+		}
+
+		// Finally, generate the encrypted packet itself.
```

Comment:
Any reason not to have this just under that existing switch statement above?

Reply:
Ah ofc since we just encrypt an empty payload if we have no special data to send.